crypto/fipsmodule/cpucap/cpucap.c

```diff
@@ -82,3 +82,6 @@ HIDDEN uint32_t OPENSSL_armcap_P = 0;
 HIDDEN uint8_t BORINGSSL_function_hit[7] = {0};
 #endif
 
+// This variable is used only for testing purposes to ensure that the library
+// constructor is executed and the capability variable is initialized.
+HIDDEN uint8_t OPENSSL_cpucap_initialized = 0;
```

Comment:
Shouldn't we turn this into a public assertion function? Ala
```
int AWSLC_is_initialized()
```
Keep the variable stuff in `crypto.o` instead of spreading it around in multiple files.

Organisationally, I think `crypto.o` should handle the init logic, so is it possible to bound it to that object?

Reply:
Also prevent us from having to add `OPENSSL_cpucap_initialized` to every new (rare, I know) cpu capability implementation.

Reply:
I thought about that and I like the idea of adding it to `crypto.o`. The problem is that then the initialization test I added in this PR would always return success, because if you are calling `AWSLC_is_initialized` which is in `crypto.o` then `crypto.o` will not be discarded. So yeah, I agree it's better to have `AWSLC_is_initialized` in `crypto.o` but I don't see a way to test this in the CI :)